staging/src/k8s.io/apiserver/pkg/endpoints/discovery/aggregated/etag.go

```diff
@@ -54,13 +54,15 @@ func ServeHTTPWithETag(
 	// Otherwise, we delegate to the handler for actual content
 	//
 	// According to documentation, An Etag within an If-None-Match
-	// header will be enclosed within doule quotes:
+	// header will be enclosed within double quotes:
 	// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/If-None-Match#directives
 	if clientCachedHash := req.Header.Get("If-None-Match"); quotedHash == clientCachedHash {
+		requestCounter.WithLabelValues(strconv.Itoa(http.StatusNotModified)).Inc()
 		w.WriteHeader(http.StatusNotModified)
 		return
 	}
 
+	requestCounter.WithLabelValues(strconv.Itoa(http.StatusOK)).Inc()
```

Comment:
I think that's fine, though I assume WRiteObjectNegotiated can decide to return a separate status code possibly if something goes wrong?

Reply:
Updated with Alex's suggestion to avoid going this route.